ekan0ra.py

```diff
@@ -123,6 +123,8 @@ def privmsg(self, user, channel, msg):
         if msg == '!' and not self.islogging:
             self.msg(self.chn, '%s no session is going on, feel free to ask a question. You do not have to type !' % user)
             return
+        if msg == '-!' and self.islogging:
+            del self.qs_queue[qs_queue.index(user)]
```

Comment:
This would remove only one occurrence. Can you refactor the code to remove all occurrences of the user?